### PR TITLE
go rewrite - Finish test template

### PR DESCRIPTION
**Pull request by c2thorn**
- Differentiates Documentation examples vs Test examples
- Adds TestCheckDestroy to test template

Generates an exact copy of `google-beta/services/datafusion/resource_data_fusion_instance_generated_test.go` except for an extra import of 
```
	"github.com/hashicorp/terraform-provider-google-beta/google-beta/envvar"
```
Which I think can be resolved by running go imports elsewhere.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/provider/template_data.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ var TemplateFunctions = template.FuncMap{
 	"camelize":   google.Camelize,
 	"underscore": google.Underscore,
 	"contains":   strings.Contains,
+	"join":       strings.Join,
```
Comment (zli82016) [inline mmv1/provider/template_data.go:54]: Sorry, I just merged my [PR](https://github.com/GoogleCloudPlatform/magic-modules/pull/10460), which added this method. But no conflict is detected and reported in this PR.

File: mmv1/api/resource/examples.go [hunk 2]
```diff
@@ -176,6 +178,32 @@ func (e *Examples) UnmarshalYAML(n *yaml.Node) error {
 }
 
 func (e *Examples) SetHCLText() {
+	e.DocumentationHCLText = ExecuteHCL(e)
+
+	copy := e
```
Comment (zli82016) [inline mmv1/api/resource/examples.go:183]: It is better to keep the comments in https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/provider/terraform/examples.rb#L204-L220?

Comment (c2thorn) [reply mmv1/api/resource/examples.go:184]: yes, added them in

File: mmv1/provider/template_data.go (hunk 1 above)
Comment (c2thorn) [reply mmv1/provider/template_data.go]: No worries. I rebased just in case.

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**